src/core/src/index/revindex/disk_revindex.rs

```diff
@@ -371,8 +379,9 @@ impl RevIndexOps for RevIndex {
 
             // TODO: Use HashesToColors here instead. If not initialized,
             //       build it.
-            match_mh
-                .iter_mins()
+            isect
+                .0
+                .iter()
```

Comment:
Nice! And with this change there is a block below:
```
counter.entry(dataset).and_modify(|e| {
    if *e > 0 {
        *e -= 1
    }
});
```
that can be
```
counter.entry(dataset).and_modify(|e| { *e -= 1 });
```
instead, and will panic if `e` goes below 0.

With this change the error triggers in `latest` in the `index::revindex::test::revindex_load_and_gather_2` test, and doesn't trigger in this PR

Reply:
added in 231a134b5, thanks!